### ISSUE TITLE
clang-tidy: modernize-use-default-member-init misses non-default constructors

### ISSUE THREAD

**Issue by ns-osmolsky** [llvm/llvm-project#52670]
Here is a trivial case that should be fixed, but is not:
```
struct PositiveNotDefaultInt {
  PositiveNotDefaultInt(int) : i(7) {}
  int i;
};
```

**Comment by ns-osmolsky** [comment]
Opened https://reviews.llvm.org/D114995 with the minimal fix. Folks have raised these two questions so far:
 1. Should there be an option to guard the extended functionality?
 2. There should be a check that we are fixing a single constructor (i.e. there is no divergence in the member's value across constructors).

**Comment by PiotrZSL** [comment]
Fixed by 051847cfecae
